File: CJLandscapeDemo2/CJLandscapeDemo2/ViewController.m
//
//  ViewController.m
//  CJLandscapeDemo2
//
//  Created by Jayxiang on 2021/3/10.
//

#import "ViewController.h"
#import "AppDelegate.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
// 在系统自动旋转的情况下测试
- (IBAction)allowRotation:(id)sender {
    ((AppDelegate *)[[UIApplication sharedApplication] delegate]).allowOrentitaionRotation = YES;
}

- (IBAction)forbidRotation:(id)sender {
    ((AppDelegate *)[[UIApplication sharedApplication] delegate]).allowOrentitaionRotation = NO;
}

@end

File: LandscapeSwift/LandscapeSwift/ViewController.swift
//
//  ViewController.swift
//  LandscapeSwift
//
//  Created by Jayxiang on 2021/7/13.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var allowButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }
    
    @IBAction func allowClick(_ sender: UIButton) {
        sender.isSelected = !sender.isSelected
        // 由于有 allowOrentitaionRotation 全局控制，BaseViewController 可以不需要
        AppDelegate.allowOrentitaionRotation = sender.isSelected
    }
    
}
